R-package/tests/testthat/test_basic.R

```diff
@@ -188,6 +188,60 @@ test_that("lightgbm() rejects negative or 0 value passed to nrounds", {
   }
 })
 
+test_that("lightgbm() accepts nrounds as either a top-level argument or parameter", {
```

Comment:
can you please add an identical test (totally fine to copy and paste the code) that this also works with `lightgbm::lgb.train()`?

This package provides both interfaces that do the same thing, mainly for backwards-compatibility reasons, so we try to test that both are behaving as expected.